lib/semantics/check-arithmeticif.cc

```diff
@@ -19,8 +19,8 @@
 
 namespace Fortran::semantics {
 
-bool IsNumericExpr(const evaluate::GenericExprWrapper &expr) {
-  auto dynamicType{expr.v.GetType()};
+bool IsNumericExpr(const SomeExpr &expr) {
```

Comment:
Thanks.  `GenericExprWrapper` usage should always be minimized, it's an ugly hack.